src/d/actor/d_a_obj_pbka.cpp

```diff
@@ -2,52 +2,133 @@
 // Generated by dtk
 // Translation Unit: d_a_obj_pbka.cpp
 //
+#include "JSystem/JKernel/JKRHeap.h"
+#include "JSystem/JUtility/JUTAssert.h"
+#include "f_op/f_op_actor_mng.h"
+#include "d/d_com_inf_game.h"
+#include "m_Do/m_Do_mtx.h"
+#include "d/d_kankyo.h"
+#include "dolphin/mtx/mtx.h"
+#include "d/d_procname.h"
 
-#include "d_a_obj_pbka.h"
-#include "dolphin/types.h"
+class daObjPbka_c : public fopAc_ac_c {
+public:
+    bool CreateHeap();
+    void CreateInit();
+    void set_mtx();
+    inline bool draw();
+public:
+    /* 0x290 */ request_of_phase_process_class mPhase;
+    /* 0x298 */ J3DModel* mpModel;
+}; //Size 0x29C
 
-/* 00000078-00000098       .text CheckCreateHeap__FP10fopAc_ac_c */
-void CheckCreateHeap(fopAc_ac_c*) {
-    /* Nonmatching */
+bool daObjPbka_c::draw()
+{
+    dKy_tevstr_c * pTevStr;
+    g_env_light.settingTevStruct(TEV_TYPE_BG0, &current.pos, pTevStr = &mTevStr);
+    g_env_light.setLightTevColorType(mpModel, pTevStr);
+    dComIfGd_setListBG();
+    mDoExt_modelUpdateDL(mpModel);
+    dComIfGd_setList();
+    return true;
+}
+
+void CheckCreateHeap(fopAc_ac_c* i_this) {
+    daObjPbka_c* a_this = (daObjPbka_c*)i_this;
+    a_this->CreateHeap();
 }
 
 /* 00000098-0000015C       .text CreateHeap__11daObjPbka_cFv */
-void daObjPbka_c::CreateHeap() {
-    /* Nonmatching */
+bool daObjPbka_c::CreateHeap() {
+    J3DModelData *modelData = (J3DModelData *)dComIfG_getObjectRes("Pbka", 3);
+    JUT_ASSERT(0x51, modelData != 0);
+    mpModel = mDoExt_J3DModel__create(modelData,0,0x11020203);
+    if(mpModel == NULL) {
+        return false;
+    }
+    return true;
 }
 
-/* 0000015C-000001CC       .text CreateInit__11daObjPbka_cFv */
 void daObjPbka_c::CreateInit() {
-    /* Nonmatching */
+    mCullMtx =  mpModel->getBaseTRMtx();
+    fopAcM_setCullSizeBox(this,-300.0,-300.0,-300.0,300.0,300.0,300.0);
+    mCullSizeFar = 0x3f800000;
```

Comment:
mCullSizeFar is a float not an int, if ghidra isn't showing it properly you can use https://gregstoll.dyndns.org/~gregstoll/floattohex/ to convert manually and figure out what the actual value is in these cases

Reply:
Nice one, thank you!